SimCalorimetry/HGCalSimProducers/src/HGCDigitizerBase.cc

```diff
@@ -51,6 +52,19 @@ HGCDigitizerBase<DFr>::HGCDigitizerBase(const edm::ParameterSet& ps) : scaleByDo
   edm::ParameterSet feCfg = myCfg_.getParameter<edm::ParameterSet>("feCfg");
   myFEelectronics_ = std::unique_ptr<HGCFEElectronics<DFr>>(new HGCFEElectronics<DFr>(feCfg));
   myFEelectronics_->SetNoiseValues(noise_fC_);
+  GenerateGaussianNoise(NoiseMean_, NoiseStd_);
+}
+
+template <class DFr>
+void HGCDigitizerBase<DFr>::GenerateGaussianNoise(const double NoiseMean, const double NoiseStd) {
+  unsigned int seed = 123456;
+  seed = seed + SeedOffset_;
+  TRandom trandom(seed);
```

Comment:
@mdhildreth I imagine that what you need here is to have a predefined noise collection before the first event noise is produced, which does not necessarily mean within the constructor. I understand the rationale behind this strategy, but I also see risks:

- is this collection big enough to avoid reusing too frequently the same noise and create artificial correlations?

- although strictly speaking your noise is reproducible reusing the same configuration, the change of random numbers among jobs in production are managed through the framework service, nobody will manually randomize the starting seed offset in your configuration, so you will end up in each single job with exactly the same noise, which I doubt is what we want. The configurable offsets will be practically useless, and you might probably structure things so as  the noise array is static since every instance will end up using the same.

Provided this is really the way you want to go I can imagine a strategy to initialise the noise array within ```runSimple``` at the first function call, to be done with care in a multi-thread safe way. That would be a clean way to rely on the random number service at least, although reusing the same library.

Reply:
@fabiocos The idea is to make the library big enough so that it doesn't matter.  It's just generating low-level single-channel noise, over millions of channels, using a gaussian distribution.  We should make it big enough so it can't be seen in physics validation plots, of course.  In terms of the random seed variation: it doesn't matter if this is randomized if the collection is big enough.  Each event gets an appropriate random entry into the library in a thread-safe way.  This should be sufficient.

We can look at runSimple to see if this is a better way.   Thanks.